ユーザー管理機能の実装

hyugayuki opened the pull request:
# WHAT
#### ユーザー管理機能をdeviseを用いて実装する
・deviseのインストール
・user model,DBの作成
・sign_upにusername保存機能を追加
・ユーザー情報編集機能の追加
・ログイン・ログアウト機能の追加
→→　フラッシュメッセージの表示機能の実装

# WHY
ログインユーザーのみがチャットを利用できるようにするため。

Commit:
create pull request

Commit:
create pull request

Commit:
Install devise

Commit:
Generate user model

Commit:
Add sign_up function

offich commented on db/schema.rb:
```diff
@@ -0,0 +1,33 @@
+    t.string   "last_sign_in_ip"
+    t.datetime "created_at",                          null: false
+    t.datetime "updated_at",                          null: false
+    t.string   "username"
```
userモデルで、`validates :username, presence: true` していますが、ここは、`null: false` のオプションをつけていただけるとベターです。

offich reviewed (changes requested):
deviseの使いかたはOKです！
1点コメントさせていただいたので、お時間よろしいときに、見ていただけると幸いです。

Commit:
edit validates

Change “presence: :sure” into “null: :false”

Commit:
Edit migraation_file

・rake db:rollback
・Edit migration_file
　　add “null: false”
・rake db:migrate
・edit user.rb

Commit:
Edit model/user.rb

Change “null: :false” into “null: false”

hyugayuki commented:
---------------　以下、ユーザー編集機能 ------------------

Commit:
Edit update_method

when update, redirect to index.haml.html

Commit:
Set up edit action

ryouzi commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the user controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
こちら使っていないのであれば削除しましょう。

ryouzi commented on db/migrate/20170908122230_add_username_to_users.rb:
```diff
@@ -0,0 +1,5 @@
+  def change
+    add_column :users, :username, :string, null: :false
+  end
+end
```
こちらのファイルはロールバックしてマイグレートしましたでしょうか？
schemaに変更がないので、気になって確認しました。

hyugayuki replied:
ターミナル上では完了できています。
<img width="1440" alt="2017-09-09 15 24 34" src="https://user-images.githubusercontent.com/31532932/30237731-10089df8-9573-11e7-95df-c1e843c97952.png">

Commit:
Delete user.scss

Unnecessary

miyagit commented on app/views/devise/mailer/confirmation_instructions.html.haml:
```diff
@@ -0,0 +1,4 @@
+%p
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/mailer/email_changed.html.haml:
```diff
@@ -0,0 +1,8 @@
+%p
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/mailer/password_change.html.haml:
```diff
@@ -0,0 +1,3 @@
+%p
+  Hello #{@resource.email}!
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/mailer/reset_password_instructions.html.haml:
```diff
@@ -0,0 +1,6 @@
+%p
+  Hello #{@resource.email}!
+%p Someone has requested a link to change your password. You can do this through the link below.
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/mailer/unlock_instructions.html.haml:
```diff
@@ -0,0 +1,5 @@
+%p
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/passwords/edit.html.haml:
```diff
@@ -0,0 +1,19 @@
+%h2 Change your password
+= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f|
+  = devise_error_messages!
+  = f.hidden_field :reset_password_token
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/passwords/new.html.haml:
```diff
@@ -0,0 +1,10 @@
+= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f|
+  = devise_error_messages!
+  .field
+    = f.label :email
```
使わないファイルは消しましょう。

miyagit commented on app/views/devise/unlocks/new.html.haml:
```diff
@@ -0,0 +1,10 @@
+= form_for(resource, as: resource_name, url: unlock_path(resource_name), html: { method: :post }) do |f|
+  = devise_error_messages!
+  .field
+    = f.label :email
```
使わないファイルは消しましょう。

Commit:
Install pry-rails

Commit:
Change pass to edit user

Commit:
Edit user_editing_function

to edit user_info without devise

hyugayuki commented:
以下、ログイン・ログアウト機能の追加

Commit:
Add sign_in/sign_out function

TOMARUMARU commented on app/controllers/messages_controller.rb:
```diff
@@ -1,4 +1,6 @@
 class MessagesController < ApplicationController
+
   def index
+    @user_id = current_user.id
```
current_user.idを定義する必要がありますでしょうか？

hyugayuki replied:
viewに設定した引数にそのままcurrent_user.idを書いて修正しました。

TOMARUMARU commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,15 @@
+  end
+
+  def update
+    current_user.update(user_params)
```
ユーザー情報編集の可否で処理を変えましょう。

Commit:
Delete unnecessary instance

Commit:
Divide update_method

TOMARUMARU commented on app/controllers/application_controller.rb:
```diff
@@ -1,3 +1,15 @@
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def after_sign_out_path_for(resource)
+    '/users/sign_in'
```
prefixで指定しましょう！

TOMARUMARU commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,18 @@
+  end
+
+  def update
+    if current_user.save(user_params)
```
saveで更新できていますでしょうか？

Commit:
Change into way_of_prefix

Commit:
Change method

Commit:
Create flash_file

Commit:
Edit to read flash file

Commit:
Delete code for design notification

Commit:
Create flash.scss

Commit:
Edit to read flash.scss

Commit:
Edit to display flash_message in japanese

Commit:
Edit to read  "devise.en.yml"

hyugayuki commented:
上記、フラッシュメッセージ機能
